src/session/session_api.c

```diff
@@ -215,7 +215,7 @@ __session_close(WT_SESSION *wt_session, const char *config)
 			WT_TRET(session->event_handler->handle_close(
 			    session->event_handler, wt_session, cursor));
 		WT_TRET(cursor->close(cursor));
-	} WT_TAILQ_SAFE_REMOVE_END;
+	}
```

Comment:
Yeah, the `else` bugs me, it feels scary (but I can't think of a bug it would introduce, either). Can you run it by Alex or Michael for a another opinion? Happy to go along with what you and they decide.